src/Symfony/Component/Messenger/Transport/RedisExt/Connection.php

```diff
@@ -169,6 +170,41 @@ public function get(): ?array
             }
         }
 
+        try {
+            // This could soon be optimized with https://github.com/antirez/redis/issues/5212 or
+            // https://github.com/antirez/redis/issues/6256
+            $pendingMessages = $this->connection->xpending($this->stream, $this->group, '-', '+', 3);
```

Comment:
If we call this on every `get` we should only read one pending message and not 3 else I would read all pending messages on first call and claim them and never call this part again unless the process is restarted.

Reply:
I do not really remember why I was switching from 1 to 3. I'll try with the new changes and it might be fine 👍